asabiris/formatter/jinja/service.py

```diff
@@ -17,18 +19,27 @@ class JinjaFormatterService(asab.Service, FormatterABC):
 
 	def __init__(self, app, service_name="JinjaService"):
 		super().__init__(app, service_name)
+		try:
+			self.Variables = {option: asab.Config.get('variables', option) for option in asab.Config.options('variables')}
+		except configparser.NoSectionError:
+			self.Variables = None
 
 
 	async def format(self, template_path, template_params):
+		if self.Variables is not None:
+			jinja_variables = collections.ChainMap(self.Variables, template_params)
```

Comment:
how about overlaps? if there is a default variable set in the configuration and the same variable is sent to iris as template_params, which should be overridden?

Reply:
@eliska-n I will try to handle that particular case

Reply:
@eliska-n In the current scenario

```
import collections

variables = {'variable1': 'config_value'}

template_params = {'variable1': 'template_param_value'}

jinja_variables = collections.ChainMap(variables, template_params)

```
print(variables['variable1'])
**Output:**
config_value

'variable1' exists in both variables and template_params, the value from variables is used.


Changing the order of dictionaries in the collections.ChainMap will affect the precedence of the variables. The order in which the dictionaries are passed to collections.ChainMap determines their precedence when resolving keys.


```
import collections

variables = {'variable1': 'config_value', 'variable2': 'config_value'}
template_params = {'variable1': 'template_param_value'}

jinja_variables = collections.ChainMap(template_params,variables)

print(variables['variable1'])
print(variables['variable2'])
```

Output:
template_param_value
config_value

template_params should override self.Variables since template_params is a set of parameters that are intended to override the default values .

Reply:
I agree.

Reply:
Yes. 

> key in template_params should override the same key in self.Variables.

This is correct.